app/domain/ptime/client.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'rest_client'
+require 'base64'
+
+module Ptime
+  class Client
+    def initialize
+      @base_url = "#{ENV.fetch('PTIME_BASE_URL')}/api/v1/"
+    end
+
+    def get(endpoint, params = {})
+      request(:get, @base_url + endpoint, params)
+    end
+
+    private
+
+    def request(method, path, params = nil)
+      path = "#{path}?#{URI.encode_www_form(params)}"
```

Comment:
don't use `path` as a variable name after passing it 
this is a URL if you ask me